### PR TITLE
Fix no overwrite alert on Windows if there are capitalization differences

### LINKED ISSUE #18354: Overwrite alert does not show when save name have different case on Windows
### Operating System

Windows 11, 64 bit

### OpenRCT2 build

OpenRCT2, v0.4.2 (8ceea45) provided by GitHub

### Describe the issue

On operating system like Windows file name like `aaa.park` and `AAA.park` is the same file.

But when you already have save names `aaa.park` and you input `AAA.park` in the save dialog it will overwrite `aaa.park` on the windows without overwrite alert shown

### Area(s) with issue?

_No response_

### Steps to reproduce

1. Play openrct2 on Windows
2. save game with name like `aaa.park`
3. start different park and save with name like `AAA.park`
4. no overwrite dialog show but `aaa.park` will be overwrite

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by jazzysoggy**
Fixes #18354

Issue: When saving on Windows, files names with only character capitalization differences, like ``aaa.park`` and ``AAA.park``, would not trigger the overwrite alert, even though Windows will overwrite the original file.

Fix: Entered in the parameter for ``String::Equals`` so that it would ignore capitalization differences if the platform is Windows. Used drive check as well as running in Wine check to make sure the platform in question is in Windows.

### CODE REVIEW

**Comment by CookiePLMonster** [comment]
Is the Wine check required? I'd hope that Wine emulates case insensitive filesystem, else breakage would be common.

**Comment by IntelOrca** [comment]
> Is the Wine check required? I'd hope that Wine emulates case insensitive filesystem, else breakage would be common.

It really depends on how Wine handles `CreateFileA` / `CreateFileW` when two files with the same name, difference case exists.

**Comment by jazzysoggy** [comment]
Removed Wine Check, since Wine has case insensitivity

File: src/openrct2-ui/windows/LoadSave.cpp [hunk 1]
```diff
@@ -609,9 +609,10 @@ static void WindowLoadsaveTextinput(rct_window* w, WidgetIndex widgetIndex, char
                 Path::Combine(_directory, text), RemovePatternWildcard(_extensionPattern));
 
             overwrite = false;
+            int32_t lowerCase = Platform::GetDrives();
```
[the remark this answers is not in the document]
Comment (CookiePLMonster) [reply src/openrct2-ui/windows/LoadSave.cpp:612]: Personally I think the check should just be done by trying to open an existing file or creating new, and reacting to a possible failure. This transfers the burden of checking the case sensitivity on the OS.

[the remark this answers is not in the document]
Comment (Gymnasiast) [reply src/openrct2-ui/windows/LoadSave.cpp:612]: That is true, but always going for case insensitivity (regardless of what the OS or filesystem does) is more user-friendly in my opinion.

[the remark this answers is not in the document]
Comment (CookiePLMonster) [reply src/openrct2-ui/windows/LoadSave.cpp:612]: I'm not sure I agree - while I do see your reasoning, I am inclined to believe going for consistency with the user's OS will always be the most user friendly.

[the remark this answers is not in the document]
Comment (jazzysoggy) [reply src/openrct2-ui/windows/LoadSave.cpp]: Implemented std::filesystem:exists check to replace the old file existence check method. This ensures consistency with underlying OS filesystem.

**Review by CookiePLMonster — CHANGES_REQUESTED** [review]
`std::filesystem::path` approach was better than this, especially since your solution will break with UTF-8 paths.

Use `File::Exists(path)`.

**Comment by jazzysoggy** [comment]
> `std::filesystem::path` approach was better than this, especially since your solution will break with UTF-8 paths.
> 
> 
> 
> Use `File::Exists(path)`.

filesystem::exists didn't seem to work with macOS and Ubuntu 18.04, the former complaining about a macOS 10.15 requirement, while the later stated such method doesn't exist.

**Comment by CookiePLMonster** [comment]
Then use `File::Exists(path)`. I suspect those environments use `ghc` for filesystem, and `File` will abstract it away from you.

**Review by CookiePLMonster — APPROVED** [review]

**Review by rik-smeets — CHANGES_REQUESTED** [review]
Could you add an entry to the changelog please? Thanks!

**Review by rik-smeets — APPROVED** [review]

**Pull request merged**